metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/store/repos/PolarisTableRepository.java

```diff
@@ -87,8 +87,9 @@ boolean existsByDbNameAndTblName(
      * @return number of rows that are updated.
      */
     @Modifying(flushAutomatically = true, clearAutomatically = true)
-    @Query("UPDATE PolarisTableEntity t SET t.metadataLocation = :newLocation, t.version = t.version + 1 "
-        + "WHERE t.metadataLocation = :expectedLocation AND t.dbName = :dbName AND t.tblName = :tableName")
+    @Query("UPDATE PolarisTableEntity t SET t.metadataLocation = :newLocation, "
+            + "t.previousMetadataLocation = t.metadataLocation, t.version = t.version + 1 "
+            + "WHERE t.metadataLocation = :expectedLocation AND t.dbName = :dbName AND t.tblName = :tableName")
```

Comment:
Recall we were discussing idempotency in the case where `previousMetadataLocation = newLocation`, whether we want it to succeed as a no op or fail. Now that `previousMetadataLocation` field is added, it is possible to add the clause `metadataLocation != newLocation` in the filter conditions if the latter behavior is more desirable.

Reply:
Discussed this offline, for now, we'll keep the existing logic of having a "retry" essentially be a no-op.